src/app/Controllers/ActorsController.cs

```diff
@@ -28,49 +31,41 @@ public ActorsController(ILogger<ActorsController> logger, IDAL dal)
         }
 
         /// <summary>
-        /// Returns a JSON array of Actor objects
+        /// Returns a JSON array of Actor objects based on query parameters
         /// </summary>
-        /// <param name="q">(optional) The term used to search Actor name</param>
-        /// <param name="pageNumber">1 based page index</param>
-        /// <param name="pageSize">page size (1000 max)</param>
-        /// <response code="200">JSON array of Actor objects or empty array if not found</response>
+        /// <param name="actorQueryParameters"></param>
         [HttpGet]
-        public async Task<IActionResult> GetActorsAsync([FromQuery] string q, [FromQuery] int pageNumber = 1, [FromQuery] int pageSize = Constants.DefaultPageSize)
+        public async Task<IActionResult> GetActorsAsync([FromQuery] ActorQueryParameters actorQueryParameters)
         {
-            string method = GetMethodText(q, pageNumber, pageSize);
-
-            // validate query string parameters
-            var result = ParameterValidator.Common(HttpContext?.Request?.Query, q, pageNumber, pageSize, method, logger);
-            if (result != null)
-            {
-                return result;
-            }
+            _ = actorQueryParameters ?? throw new ArgumentNullException(nameof(actorQueryParameters));
 
             // convert to zero based index
-            pageNumber = pageNumber > 1 ? pageNumber - 1 : 0;
+            actorQueryParameters.PageNumber = actorQueryParameters.PageNumber > 1 ? actorQueryParameters.PageNumber - 1 : 0;
 
-            return await ResultHandler.Handle(dal.GetActorsAsync(q, pageNumber * pageSize, pageSize), method, Constants.ActorsControllerException, logger).ConfigureAwait(false);
+            return await ResultHandler.Handle(dal.GetActorsAsync(
+                    actorQueryParameters.Q,
```

Comment:
why not simplify GetActors and pass the class instead of the individual values?

Reply:
This requires changing this code, the interface for IDal, and the CosmosHealthCheck code which are not related to the original request. I would rather focus on the http/400 response issue and look to optimize this code at a later time if it makes sense.

Reply:
I would fix it for completeness and clarity, but understand wanting to close the PR. Your call.

Reply:
@bartr this has been fixed in the latest commit. It was relatively easy to implement and the code does look more clear now.